### DISCUSSION TITLE
Flexible Client/Server model.

### DISCUSSION THREAD

**Discussion by fullstopslash** [debauchee/barrier#1581] [Ideas]
I've had this question lurking in my mind. Is it currently possible to use barrier in such a way that each system can act simultaneously as a client /and/ a server. Basically I have a couple of systems and multiple keyboards and It'd be nice to just use whichever one I'm closest to at the moment, not needing a dedicated server.

**Comment by matteote** [comment]
I would love to have this feature. I would like to replicate on macOS what Mouse Without Borders does on Windows, allowing to use mouse and keyboard from any of the machines.

**Comment by leonjza** [comment]
I have also been thinking about a variation of this request for a while. In my use case, I'd love to be able to configure Barrier to use the macOS server's keyboard, but a Windows clients mouse across both hosts.

**Comment by 47k** [comment]
I really appreciate this feature! I between two setups: 
-at the day I'm sitting on my MAC and control with my MAC my Windows PC 
-in the evening I'm sitting on my Windows PC and want control my MAC with my Windows PC 

Two options are fine for me:
1. To use both sides without any special switwiching - depending on which inputs I'm using 
2. A simple button to switch between Client and Server

**Comment by fullstopslash** [comment]
Truly, I think it would fundamentally require a different model. instead of being a client/server application you'd enroll computers in a distributed input stream. How would one prioritize which input gets priority if two inputs are active at the same time? I get that it'd be a rather hard problem to solve.
